camel/utils/token_counting.py

```diff
@@ -211,10 +215,59 @@ def count_tokens_from_messages(self, messages: List[OpenAIMessage]) -> int:
         for message in messages:
             num_tokens += self.tokens_per_message
             for key, value in message.items():
-                num_tokens += len(self.encoding.encode(str(value)))
+                if not isinstance(value, list):
+                    num_tokens += len(self.encoding.encode(str(value)))
+                else:
+                    for item in value:
+                        if item["type"] == "text":
+                            num_tokens += len(
+                                self.encoding.encode(str(item["text"])))
+                        elif item["type"] == "image_url":
+                            image_str: str = item["image_url"]["url"]
+                            detail = item["image_url"]["detail"]
+                            image_prefix_format = "data:image/{};base64,"
+                            image_prefix: Optional[str] = None
+                            for image_type in list(OpenAIImageType):
```

Comment:
Why `optional`, and since you want to assert the instance `assert isinstance(image_prefix, str)`?

Reply:
Initialize with `None` as there is no good value to initialize. Following `assert isinstance(image_prefix, str)` is to pass the format checking.